[rpc] fix test_debug_info for python 3.5

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#34828 [rpc] fix test_debug_info for python 3.5**

python 3.5 does not ensure ordering of dictionary keys, this was added
in python 3.6+. Fixing this so the test is no longer flaky in 3.5. Tested by
500 stresstests with python 3.5

Differential Revision: [D20474996](https://our.internmc.facebook.com/intern/diff/D20474996/)

rohan-varma committed:
[rpc] fix test_debug_info for python 3.5

python 3.5 does not ensure ordering of dictionary keys, this was added
in python 3.6+. Fixing this so the test is no longer flaky in 3.5. Tested by
500 stresstests with python 3.5

Differential Revision: [D20474996](https://our.internmc.facebook.com/intern/diff/D20474996/)

[ghstack-poisoned]

mrshenli reviewed (approved):
Thanks for the fix!!

rohan-varma committed:
Update on "[rpc] fix test_debug_info for python 3.5"


python 3.5 does not ensure ordering of dictionary keys, this was added
in python 3.6+. Fixing this so the test is no longer flaky in 3.5. Tested by
500 stresstests with python 3.5

Differential Revision: [D20474996](https://our.internmc.facebook.com/intern/diff/D20474996/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[rpc] fix test_debug_info for python 3.5"


python 3.5 does not ensure ordering of dictionary keys, this was added
in python 3.6+. Fixing this so the test is no longer flaky in 3.5. Tested by
500 stresstests with python 3.5

Differential Revision: [D20474996](https://our.internmc.facebook.com/intern/diff/D20474996/)

[ghstack-poisoned]

rohan-varma commented:
Confirmed by printing out the keys, that on 3.5, the key order is sometimes different, hence failing the test:

```
AssertionError: 'num_owner_rrefs' != 'num_pending_users'
- num_owner_rrefs
+ num_pending_users
 : expected keys are dict_keys(['num_autograd_contexts', 'agent.thread_pool_size', 'num_owner_rrefs', 'num_pending_users', 'agent.num_idle_threads', 'agent.server_active_async_calls', 'agent.client_active_calls', 'agent.server_active_calls', 'num_current_backward_passes', 'local_autograd_engine_cpu_queue_size', 'agent.num_pending_requests']), info keys are dict_keys(['num_autograd_contexts', 'agent.thread_pool_size', 'num_pending_users', 'num_owner_rrefs', 'agent.num_idle_threads', 'agent.server_active_async_calls', 'agent.client_active_calls', 'agent.server_active_calls', 'num_current_backward_passes', 'local_autograd_engine_cpu_queue_size', 'agent.num_pending_requests'])
```
